### PR TITLE
ddl: add COMMENTS column to DDL jobs and enhance job reorg meta handling

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/57229 #57526

Problem Summary: See #57229

### What changed and how does it work?

- Add `COMMENTS` column to `ADMIN SHOW DDL JOBS`.
- Refine DDL job reorg meta initialization. For example, let "modify column"/"reorg partition" and other reorg DDL types utilize `ReorgMeta.Concurrency` and `ReorgMeta.BatchSize`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
add COMMENTS column to ADMIN SHOW DDL JOBS
```

### CODE REVIEW

File: pkg/meta/model/job.go [hunk 1]
```diff
@@ -492,6 +493,36 @@ func (job *Job) FillFinishedArgs(args FinishedJobArgs) {
...
+			}
+			if m.UseCloudStorage {
+				labels = append(labels, "cloud")
+			}
+		case ReorgTypeTxnMerge:
+			labels = append(labels, ReorgTypeTxnMerge.String())
+		}
+	}
+	if job.MayNeedReorg() {
+		labels = append(labels, fmt.Sprintf("concurrency=%d", m.Concurrency))
```
Comment (tangenta) [inline pkg/meta/model/job.go:520]:
```suggestion
		labels = append(labels, fmt.Sprintf("thread=%d", m.Concurrency))
```

File: pkg/ddl/executor.go [hunk 2]
```diff
@@ -4937,44 +4915,81 @@ func (e *executor) createIndex(ctx sessionctx.Context, ti ast.Ident, keyType ast
...
+				err := setDistTaskParam()
+				if err != nil {
+					return err
+				}
+			case model.ActionReorganizePartition,
+				model.ActionRemovePartitioning,
+				model.ActionAlterTablePartitioning,
+				model.ActionModifyColumn:
+				setReorgParam()
+				continue
```
Comment (fzzf678) [inline pkg/ddl/executor.go:4975]:
```suggestion
```
This continue seems can be removed

**Review by fzzf678 — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**